utils/spdx/src/main/kotlin/SpdxExpression.kt

```diff
@@ -244,6 +249,21 @@ class SpdxCompoundExpression(
     override fun normalize(mapDeprecated: Boolean) =
         SpdxCompoundExpression(operator, children.map { it.normalize(mapDeprecated) })
 
+    override fun simplify(): SpdxExpression {
+        val flattenedChildren = children.flatMapTo(mutableSetOf()) { child ->
+            val simplifiedChild = child.simplify()
+
+            if (simplifiedChild is SpdxCompoundExpression && simplifiedChild.operator == operator) {
+                // Inline nested children of the same operator.
+                simplifiedChild.children.map { it.simplify() }
+            } else {
+                setOf(simplifiedChild)
+            }
+        }
+
+        return SpdxCompoundExpression(operator, flattenedChildren)
```

Comment:
Maybe this function should be called by `normalize` as well. Also, this will throw an exception if `flattenedChildren.size < 2`.

Reply:
> Maybe this function should be called by `normalize` as well.

I have a slight tendency towards keeping normalizing and simplifying separate use-cases. 

However, having the respective `File` functions in mind, we should have probably called the current `normalize` rather `canonize`, leaving room for calling this function `normalize` instead of `simplify`.

> Also, this will throw an exception if flattenedChildren.size < 2.

Good find. Fixed, and added a test for that case.